otcextensions/osclient/ces/v1/alarm.py

```diff
@@ -0,0 +1,424 @@
+            metavar='<alarm_action_notification_list>',
+            action='append',
+            help=_('Specifies the list of objects being notified when '
+                   'alarm status changes.\n'
+                   'URN example structure:\n'
+                   'urn:smn:region:68438a86d98e427e907e0097b7e35d48:sd\n'
+                   'The parameter can be given multiple times to '
+                   'notify multiple targets.')
+        )
+
+        # OkActions
+        parser.add_argument(
+            '--ok-action-type',
+            metavar='<ok_action_type>',
+            help=_('Specifies the alarms action type.\n'
+                   'notification: notification will be sent to user\n'
+                   'autoscaling: scaling action will be triggered')
+        )
+        parser.add_argument(
+            '--ok-action-notification-list',
+            metavar='<ok_action_notification_list>',
+            action='append',
+            help=_('Specifies the list of objects being notified when '
+                   'alarm status changes.\n'
+                   'URN example structure:\n'
+                   'urn:smn:region:68438a86d98e427e907e0097b7e35d48:sd\n'
+                   'The parameter can be given multiple times to '
+                   'notify multiple targets.')
+        )
+
+        # ConditionSpec
+        parser.add_argument(
+            '--comparison-operator',
+            metavar='<comparison_operator>',
+            required=True,
+            help=_('Specifies the conditions comparison operator')
+        )
+        parser.add_argument(
+            '--count',
+            metavar='<count>',
+            type=int,
+            required=True,
+            help=_('Specifies how many times the alarm condition has to '
+                   'triggered until Alarm raises.\n'
+                   'Value range: 1 to 5')
+        )
+        parser.add_argument(
+            '--filter',
+            metavar='<filter>',
+            required=True,
+            help=_('Specifies the data rollup method.\n'
+                   'Values: max, min, average, sum, variance')
+        )
+        parser.add_argument(
+            '--period',
+            metavar='<period>',
+            type=int,
+            required=True,
+            help=_('Indicates the interval (in seconds) for checking '
+                   'whether the configured alarm rules are met.')
+        )
+        parser.add_argument(
+            '--unit',
+            metavar='<unit>',
+            help=_('Specifies data unit.')
+        )
+        parser.add_argument(
+            '--value',
+            metavar='<value>',
+            type=int,
+            required=True,
+            help=_('Specifies the alarm threshold.\n'
+                   'Values: 0 to max(int)')
+        )
+
+        # DimensionsSpec for Metrics
+        # This is a list of dictionaries
+        # IMPROVEMENT NEEDED
+        parser.add_argument(
+            '--dimension-name',
```

Comment:
I think --dimension-name and --dimension-value parser argument can be removed and  then add_arguments as --dimensions,   the action will become "action=parseractions.MultiKeyValueAction
To get result -- 
```
"dimensions": [
            {
                "name": "instance_id", 
                "value": "xxxxxxx-xxxx-xxxxxxxxx-xxxx"
            }
        ], 
```
Refer - https://docs.openstack.org/osc-lib/pike/reference/api/osc_lib.cli.parseractions.html

Reply:
Hi Vineet, thx for your hints. I try to improve this